mlrun/datastore/sources.py

```diff
@@ -206,12 +208,23 @@ def __init__(
         name: str = None,
         path: str = None,
         attributes: Dict[str, str] = None,
+        key_field: str = None,
+        time_field: str = None,
         workers: int = None,
     ):
-        super().__init__(name, path, attributes)
+        super().__init__(name, path, attributes, key_field, time_field)
         self.online = True
         self.workers = workers
 
+    def to_step(self, key_field=None, time_field=None):
+        import storey
+
+        return storey.Source(
+            key_field=self.key_field or key_field,
```

Comment:
Shouldn't be the opposite ?
if we given a value use it, and if not use the `self.x` ?

Reply:
no, the target class wins over the default which comes from the feature set spec